### PR TITLE
Search for team names too

### PR DESCRIPTION
**Pull request by F3PiX**
#379 
The search form now returns team names too.
- [x] Query for searching team names
- [x] Search result with both users with matching names and users with matching team names
- [x] Make class method work from outside controller
- [x] Write tests

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -155,6 +155,7 @@ def current_student?
 
   def self.search(search)
     where("users.name ILIKE ?", "%#{search}%")
+    where("teams.name ILIKE ?", "%#{search}%")
```
Comment (carpodaster) [inline app/models/user.rb]: Hey @F3PiX, I think chaining `where` statements constructs `AND` queries in SQL. Furthermore, I believe the teams association has to be added with `includes(:teams).references(:teams)`, too.

Comment (F3PiX) [inline app/models/user.rb]: Thank you! That helps.

**Comment by F3PiX** [comment]
The good news is: it works. 
The query returns both user names and team names. 
find users and teams
I don't understand WHY it works with `teams.name`. (Could it be because of the `has_many`-relations? I trial-and-errored with `roles.name` and that works as well. )
And if it DOES work, is it an acceptable way as well?

**Comment by carpodaster** [comment]
Morning @F3PiX 

>  I don't understand WHY it works with teams.name.

It works in the controller, because line 16 states `with_all_associations_joined`. Try firing `User.search "cheesy"` in your rails console to reproduce the error I was expecting in an earlier comment.

Since `User.search` is part of the public API of that model and therefore can be called out of the controller context, it should bring its own include/join statements. We should probably have a unit test for the class method as well.

**Comment by F3PiX** [comment]
Aha. 
I need some time figuring out how to put the associations into the combined query. I would like to, it's something I want to find out and I have an action plan. 🚀

File: app/views/users/index.html.slim [hunk 2]
```diff
@@ -41,5 +41,5 @@ table.users.table.table-striped.table-bordered.table-condensed.table-hover
         / td = links_to_conferences(user.conferences.current).join(', ').html_safe
 
 p.pagination-info
-  = page_entries_info @users
-  = paginate @users, theme: 'twitter-bootstrap-3', pagination_class: 'pagination-sm'
+  = page_entries_info @users unless params[:search].present?
```
Comment (F3PiX) [inline app/views/users/index.html.slim]: Tips for better solution than this?

**Comment by F3PiX** [comment]
Hi @ramonh. Could you take a look please?

File: app/views/users/index.html.slim (hunk 2 above)
Comment (F3PiX) [inline app/views/users/index.html.slim]: I found a better solution. Following in next pr.

**Comment by ramonh** [comment]
Looks great, thank you @F3PiX! ✌🏽

**Pull request merged**